compiler/ballerina-compiler-api/src/main/java/io/ballerina/compiler/api/symbols/FunctionSymbol.java

```diff
@@ -19,27 +19,26 @@
 
 import io.ballerina.compiler.api.types.BallerinaTypeDescriptor;
 
-import java.util.List;
 import java.util.Optional;
 
 /**
  * Represent Function Symbol.
  *
  * @since 2.0.0
  */
-public interface FunctionSymbol extends Symbol {
+public interface FunctionSymbol extends Symbol, Qualifiable, Deprecatable {
 
     /**
-     * Get the list of qualifiers.
+     * Get the Function Type Descriptor.
      *
-     * @return {@link List} of qualifiers
+     * @return {@link BallerinaTypeDescriptor}
      */
-    List<Qualifier> qualifiers();
+    Optional<BallerinaTypeDescriptor> typeDescriptor();
```

Comment:
Can the type-desc be null?

Reply:
Hmm. Can't think of a reason why it'd be. Will check with Nadeeshaan and see. All the `typeDescriptor()` methods seem to return `Optional` values.

Reply:
This was because the earlier BOperatorSymbols were converted into the InvokableSymbols and the type became null during the conversion, I think we can remove the optional here